### ISSUE TITLE
scheduleNextRecurrence function called more than 1400 times

### ISSUE THREAD

**Issue by ytkang** [node-schedule/node-schedule#214]
### Codes
```javascript
schedule.scheduleJob('0 * * * * *', function(){console.log("*** recurrence ***");});
schedule.scheduleJob("Should be One shot", new Date(Date.now() + 24*60*60*1000), function(){console.log("*** Should be One shot! ***");});
```

### Explanation
If you execute above codes and **change machine time after 1day**,
Then, the **"recurrence"** message will be printed more than **1400 times** and **"Should be One shot"** will be printed **2 times**.

**Comment by santigimeno** [comment]
@ytkang it looks the problem is similar to the one described here: https://github.com/node-schedule/node-schedule/issues/208
I don't think there's too much we can do about it. Well maybe there is a way but I don't know of it

**Comment by ytkang** [comment]
@santigimeno  yeah, I think It is almost same issue, but It seems there is no second case about "Should be One Shot". I hope me or someone could find solution! to make more strong library.

**Comment by santigimeno** [comment]
@ytkang yes, I think the `one shot case` might be fixed. I'll try to take a look at it. In order to reproduce the case exactly: can you tell me the date you change the machine date to?

**Comment by ytkang** [comment]
@santigimeno Sure, 
Because the time was **new Date(Date.now() + 24*60*60*1000)**
so you need to change your machine time as current time +  1day (24*60*60*1000)

**Comment by ytkang** [comment]
@santigimeno, I just tried to fix this issue. please review this.

**Comment by ans-4175** [comment]
hey, it is still occurred. I happened use it perfectly before change my timezone data, then after I changed it, it suddenly stop working. Any ideas?

**Comment by tejasmanohar** [comment]
@ans-4175 If the process restarts in b/w your system time change, there shouldn't be a problem since node-schedule is for the most part composed of pure functions and doesn't really persist anything.

Let me know if I'm missing something @santigimeno @ytkang

**Comment by ytkang** [comment]
@ans-4175 There was an bug stop working when you use cron-style schedule,
see this issue. https://github.com/node-schedule/node-schedule/pull/233

**Comment by ans-4175** [comment]
yes @ytkang I use cron-style schedule '00 17 * * *', I want to execute function/job every day at 5pm UTC. Before that my I wrote '00 12 * * *' and tzdata configured in EST (UTC -5), then I changed my tzdata to UTC. @tejasmanohar. But turned out it didnt work. I'll take look at this issue. Is it resolved ?

**Comment by nrempel** [comment]
Hey there, I'm playing with this bug and have a slightly different take on it.

I think that jobs that occur that during a system time change should be missed completely. In this case, I think that the "one shot" job should be run 0 times (because the scheduled time effectively never happened according to system time).

What do you think about this? I believe this is consistent with how the Cron unix utility handles this case.

**Comment by tejasmanohar** [comment]
I agree that we should stay consistent with `cron` behavior and not something like `anacron`. All these system changes are so system specific that I think they should be handled by whoever maintains the system / code running on it :)

**Comment by nrempel** [comment]
Thanks for the response!

I agree that system changes should be handled by the system maintainer but I _do_ think that the "catching up" behaviour we see right now should be considered a bug. Also, the one-shot job should definitely not be run twice.

By agreeing, can I assume you will accept a pull request to drop invocations that occur in the past? 😀
